lib/ansible/module_utils/docker_common.py

```diff
@@ -381,7 +382,7 @@ def find_image(self, name, tag):
 
         self.log("Find image %s:%s" % (name, tag))
         images = self._image_lookup(name, tag)
-        if len(images) == 0:
+        if len(images) == 0 and name not in DOCKER_RESERVED_IMAGE_NAMES:
             # In API <= 1.20 seeing 'docker.io/<name>' as the name of images pulled from docker hub
```

Comment:
`sha256` is reserved by docker, it's good enough to detect the usage of hash id.